semgrep-core/bin/Main.ml

```diff
@@ -264,6 +275,55 @@ let gen_layer ~root ~query file =
   ()
 (*e: function [[Main_semgrep_core.gen_layer]] *)
 
+let filemtime file =
+  (Unix.stat file).Unix.st_mtime
+
+(* The function below is mostly a copy-paste of Common.cache_computation.
+ * This function is slightly more flexible because we can put the cache file
+ * anywhere thanks to the argument 'cache_file_of_file'.
+ * We also try to be a bit more type-safe by using the version tag above.
+ * TODO: merge in pfff/commons/Common.ml at some point
+ *)
+let cache_computation file cache_file_of_file f =
+  if not !use_parsing_cache
+  then f ()
+  else begin
+    if not (Sys.file_exists file)
+    then begin
+      pr2 ("WARNING: cache_computation: can't find file "  ^ file);
+      pr2 ("defaulting to calling the function");
+      f ()
+    end else begin
+    profile_code "Main.cache_computation" (fun () ->
+
+      let file_cache = cache_file_of_file file in
+      if Sys.file_exists file_cache && filemtime file_cache >= filemtime file
```

Comment:
Not sure how Docker cp works, but just flagging that this may always be invalid in that case.